syncserver-db-common/src/error.rs

```diff
@@ -8,14 +8,14 @@ use thiserror::Error;
 /// Error specific to any MySQL database backend. These errors are not related to the syncstorage
 /// or tokenserver application logic; rather, they are lower-level errors arising from diesel.
 #[derive(Debug)]
-pub struct MysqlError {
-    kind: MysqlErrorKind,
+pub struct SqlError {
```

Comment:
Maybe I can split that into a separate PR that can get merged before this one.
Because I really think that the renaming should be done before adding the new backend.
If I do the renaming later I will mess up things and I won't know if something is MySQL specific or generic.

Reply:
Yeah, I do think the rename makes a lot of sense, particularly if we're going to have more than one SQL backend. 
I'm fine with boosting the rename PR since it'll also reduce the size and complexity of this one. 

(Granted, I've also been debating making the default build "spanner", since that's what we use for production, but that's beside the point.)

Reply:
Here's the new PR #1619 
It's based of the master branch. (I can revert the second commit if needed but that was triggering clippy)